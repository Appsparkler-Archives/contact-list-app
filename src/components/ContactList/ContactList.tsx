import * as React from "react";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import Divider from "@mui/material/Divider";
import ListItemText from "@mui/material/ListItemText";
import Typography from "@mui/material/Typography";
import { TContactListFC } from "types";

export const ContactList: TContactListFC = ({ contacts }) => {
  return (
    <List sx={{ width: "100%", bgcolor: "background.paper" }}>
      {contacts.map((contact) => (
        <React.Fragment key={contact.id}>
          <ListItem alignItems="center">
            <ListItemText
              primary={contact.name}
              secondary={
                contact.company && (
                  <Typography
                    sx={{ display: "inline" }}
                    component="span"
                    variant="body2"
                    color="text.primary"
                  >
                    {contact.company}
                  </Typography>
                )
              }
            />
          </ListItem>
          <Divider variant="fullWidth" component="li" />
        </React.Fragment>
      ))}
    </List>
  );
};
