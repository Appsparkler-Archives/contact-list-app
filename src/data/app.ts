import { IContactFormData } from "types";

export type TAppFC = React.FC<{
  contacts: IContactFormData[];
}>;
