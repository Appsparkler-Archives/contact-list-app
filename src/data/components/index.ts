export * from "./standard-select-field";
