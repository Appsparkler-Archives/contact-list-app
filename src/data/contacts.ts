import { uniqueId } from "lodash/fp";
import { IContactFormData } from "types";

export const contactsData: Record<"variant1", IContactFormData[]> = {
  variant1: [
    {
      id: "7cb1248c-983a-4207-a59f-40bfa1fae564",
      name: "Ali Connors",
      gender: "male",
      mobile: "+9282822223",
      email: "ali.connors@whatsapp.com",
      type: "business",
      imgUrl:
        "https://plus.unsplash.com/premium_photo-1671641797679-3b680a7d2109?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8cHJvZmlsZSUyMHBpY3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=900&q=60",
      company: "SurveyHeart",
      address: "UK",
    },
    {
      id: "b4fc7a3e-3522-4292-b05d-5edbdffcf3ac",
      name: "Jenny Allen",
      gender: "female",
      mobile: "+39292323",
      type: "personal",
      imgUrl:
        "https://images.unsplash.com/photo-1625621435205-2d09ed661a71?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=987&q=80",
      company: "Societe Generale",
    },
    {
      id: "e90ae941-f295-4adc-bf8a-b1d7127a60e6",
      gender: "male",
      mobile: "+9211111214343",
      type: "business",
      imgUrl:
        "https://images.unsplash.com/photo-1500648767791-00dcc994a43e?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8cHJvZmlsZSUyMHBpY3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=900&q=60",
      name: "Steven Covey",
    },
  ],
};

export const getDefaultCreateContact = (): IContactFormData => ({
  id: uniqueId("contact"),
  gender: "female",
  mobile: "",
  name: "",
  type: "personal",
  address: "",
  company: "",
  email: "",
});
