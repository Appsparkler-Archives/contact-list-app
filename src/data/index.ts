export * from "./contacts";
export * from "./components";
