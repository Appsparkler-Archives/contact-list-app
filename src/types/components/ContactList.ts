import React from "react";
import { IContactFormData } from "types";

export type TContactListFC = React.FC<{ contacts: IContactFormData[] }>;
