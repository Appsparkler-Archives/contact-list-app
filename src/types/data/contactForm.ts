export interface IContactForm {
  name: string;
  mobile: string;
  address: string;
  company: string;
  email: string;
  gender: "male" | "female";
  type: "personal" | "business";
}
