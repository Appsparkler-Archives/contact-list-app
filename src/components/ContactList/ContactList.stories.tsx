import type { Meta, StoryObj } from "@storybook/react";
import { ContactList } from "./ContactList";
import { contacts } from "data";

const meta = {
  title: "Components/Contact List",
  component: ContactList,
  tags: ["components"],
} satisfies Meta<typeof ContactList>;

export default meta;

type Story = StoryObj<typeof meta>;

export const contactList: Story = {
  args: {
    contacts,
  },
};
