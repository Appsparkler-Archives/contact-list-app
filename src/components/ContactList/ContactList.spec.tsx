import renderer from "react-test-renderer";
import { contactsData } from "data";
import { ContactList } from "./ContactList";

describe("ContactList", () => {
  test("snapshot test", async () => {
    const component = renderer.create(
      <ContactList contacts={contactsData.variant1} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
