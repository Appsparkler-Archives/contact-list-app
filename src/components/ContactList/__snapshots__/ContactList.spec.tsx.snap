// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList snapshot test 1`] = `
<ul
  className="MuiList-root MuiList-padding css-168pop2-MuiList-root"
>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Ali Connors
      </span>
      <span
        className="MuiTypography-root MuiTypography-body2 css-1phucj-MuiTypography-root"
      >
        SurveyHeart
      </span>
    </div>
  </li>
  <li
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    role="separator"
  />
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
  >
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Jenny Allen
      </span>
      <span
        className="MuiTypography-root MuiTypography-body2 css-1phucj-MuiTypography-root"
      >
        Societe Generale
      </span>
    </div>
  </li>
  <li
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    role="separator"
  />
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Steven Covey
      </span>
    </div>
  </li>
  <li
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    role="separator"
  />
</ul>
`;
